src/common/models/project.go

```diff
@@ -103,12 +164,9 @@ type BaseProjectCollection struct {
 
 // ProjectRequest holds informations that need for creating project API
 type ProjectRequest struct {
-	Name                                       string `json:"project_name"`
-	Public                                     int    `json:"public"`
-	EnableContentTrust                         bool   `json:"enable_content_trust"`
-	PreventVulnerableImagesFromRunning         bool   `json:"prevent_vulnerable_images_from_running"`
-	PreventVulnerableImagesFromRunningSeverity string `json:"prevent_vulnerable_images_from_running_severity"`
-	AutomaticallyScanImagesOnPush              bool   `json:"automatically_scan_images_on_push"`
+	Name     string            `json:"project_name"`
+	Public   *int              `json:"public"` //deprecated, reserved for project creation in replication
```

Comment:
Please raise issue for: "Define a compatible checking interface and implement it in the api layer function if that API needs "